### PR TITLE
added slack duplicate user warning

### PR DESCRIPTION
**Pull request by bprize15**
This fixes https://github.com/oncokb/oncokb/issues/2664

### CODE REVIEW

File: src/main/java/org/mskcc/cbio/oncokb/service/UserService.java [hunk 1]
```diff
@@ -667,6 +668,23 @@ public boolean userHasAuthority(User user, String authority) {
         return user.getAuthorities().stream().filter(userAuth -> userAuth.getName().equalsIgnoreCase(authority)).count() > 0;
     }
 
+    public List<UserDTO> getPotentialDuplicateAccountsByUser(UserDTO user) {
+        return searchAccountsForPotentialDuplicateUser(user, getAllManagedUsers());
+    }
+
+    public List<UserDTO> searchAccountsForPotentialDuplicateUser(UserDTO user, List<UserDTO> allUsers) {
+        String userFullName = user.getFirstName() + " " + user.getLastName();
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/service/UserService.java:676]: Can we make this a method `getFullName`?

File: src/main/java/org/mskcc/cbio/oncokb/service/UserService.java [hunk 2]
```diff
@@ -667,6 +668,23 @@ public boolean userHasAuthority(User user, String authority) {
...
+        JaroWinklerSimilarity jw = new JaroWinklerSimilarity();
+        List<UserDTO> potentialDuplicateUsers = new ArrayList<>();
+        for (UserDTO potentialDuplicate : allUsers) {
+            String potentialDuplicateFullName = potentialDuplicate.getFirstName() + " " + potentialDuplicate.getLastName();
+            if (user.getId() != potentialDuplicate.getId() && jw.apply(userFullName, potentialDuplicateFullName) > .7) {
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/service/UserService.java:681]: Does the params order matter in  `jw.apply`?

**Review by zhx828 — CHANGES_REQUESTED** [review]

File: src/main/java/org/mskcc/cbio/oncokb/service/UserService.java (hunk 2 above)
Comment (bprize15) [reply src/main/java/org/mskcc/cbio/oncokb/service/UserService.java:681]: Nope

File: src/main/java/org/mskcc/cbio/oncokb/service/UserService.java [hunk 3]
```diff
@@ -667,6 +668,23 @@ public boolean userHasAuthority(User user, String authority) {
...
+    public List<UserDTO> getPotentialDuplicateAccountsByUser(UserDTO user) {
+        return searchAccountsForPotentialDuplicateUser(user, getAllManagedUsers());
+    }
+
+    public List<UserDTO> searchAccountsForPotentialDuplicateUser(UserDTO user, List<UserDTO> allUsers) {
+        String userFullName = StringUtil.getFullName(user.getFirstName(), user.getLastName());
+        JaroWinklerSimilarity jw = new JaroWinklerSimilarity();
+        List<UserDTO> potentialDuplicateUsers = new ArrayList<>();
+        for (UserDTO potentialDuplicate : allUsers) {
+            String potentialDuplicateFullName = potentialDuplicate.getFirstName() + " " + potentialDuplicate.getLastName();
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/service/UserService.java:680]: @bprize15 could you update using `getFullName` here as well?

**Review by zhx828 — APPROVED** [review]
LGTM. Thanks! Don't forget to squash before merging. If you don't know how, let me know.

**Pull request merged**